pool/pool.go

```diff
@@ -60,8 +60,7 @@ type clientStatus interface {
 	address() string
 	currentErrorRate() uint32
 	overallErrorRate() uint64
-	latency() time.Duration
-	requests() uint64
+	methodsStatus() map[string]methodStatus
```

Comment:
It is finite and fixed in size, can we use a slice with `method*` constants being indices?
The only thing where a map gives something is `incRequests`, but again, all methods are known in advance, so we can preallocate everything.